com.unity.mobile.notifications/Runtime/Android/Plugins/com/unity/androidnotifications/UnityNotificationUtilities.java

```diff
@@ -31,30 +47,371 @@ protected static int findResourceIdInContextByName(Context context, String name)
     }
 
     protected static String serializeNotificationIntent(Intent intent) {
-        Bundle bundle = intent.getExtras();
+        try {
+            Notification notification = intent.getParcelableExtra("unityNotification");
+            if (notification == null)
+                return null;
+            ByteArrayOutputStream data = new ByteArrayOutputStream();
+            DataOutputStream out = new DataOutputStream(data);
+            boolean success = serializeNotificationParcel(intent, out);
+            if (!success) {
+                data.reset();
+                success = serializeNotificationCustom(notification, out);
+            }
+            if (success) {
+                out.close();
+                byte[] bytes = data.toByteArray();
+                return Base64.encodeToString(bytes, 0, bytes.length, 0);
+            }
+        } catch (Exception e) {
+            Log.e("Unity", "Failed to serialize notification", e);
```

Comment:
Wondering if we can output some useful info like notification id etc. here, it can help user to identify which notification is causing problem.